osu.Game/Screens/Select/Carousel/CarouselBeatmapSet.cs

```diff
@@ -99,6 +99,12 @@ public override int CompareTo(FilterCriteria criteria, CarouselItem other)
 
                 case SortMode.Difficulty:
                     return compareUsingAggregateMax(otherSet, b => b.StarRating);
+
+                case SortMode.DateSubmitted:
+                    if (BeatmapSet.DateSubmitted == null || otherSet.BeatmapSet.DateSubmitted == null)
+                        return 0;
```

Comment:
This is not going to be correct. You've copied this from above, but removed/ignored the comment explaining why it *can* exist like this.

https://github.com/ppy/osu/blob/a05d7f4d8ca28dd7022ee94d280e267133d7e174/osu.Game/Screens/Select/Carousel/CarouselBeatmapSet.cs#L85

https://github.com/ppy/osu/blob/438067a18b4748501c49a57db0201212eb22829c/osu.Game/Screens/Select/Carousel/CarouselBeatmap.cs#L58

Note that in the other usage, beatmap sets with missing date are hidden. I think that should also be the case here, so similar hiding logic will need to be added.

Also I'd argue the filtering should be done in `CarouselBeatmapSet`'s `Filter` override, since both the linked case and this case are operating on sets rather than individual difficulties. The existing case can be moved at the same time as adding the new one.

If you aren't sure what I mean, I can make the change for you.

Reply:
Hi peppy, I would like to try again. Sorry.